vortex-layout/src/scan/task.rs

```diff
@@ -36,17 +35,14 @@ impl ScanTask {
     }
 }
 
-/// A trait used to spawn and execute blocking tasks.
-#[async_trait]
 pub trait TaskExecutor: 'static + Send + Sync {
-    async fn execute(&self, array: &Array, tasks: &[ScanTask]) -> VortexResult<Array>;
+    fn execute(&self, array: &Array, tasks: &[ScanTask]) -> VortexResult<Array>;
 }
 
 pub struct InlineTaskExecutor;
 
-#[async_trait]
 impl TaskExecutor for InlineTaskExecutor {
-    async fn execute(&self, array: &Array, tasks: &[ScanTask]) -> VortexResult<Array> {
+    fn execute(&self, array: &Array, tasks: &[ScanTask]) -> VortexResult<Array> {
```

Comment:
this is a flyby but seems like it can be blocking now? potentially not even needed but I'm not sure what are your plans for it